packages/test/test-end-to-end-tests/src/test/gc/gcTestSummaryUtils.ts

```diff
@@ -0,0 +1,179 @@
+    }
+
+    const request: IRequest = {
+        headers: {
+            [LoaderHeader.cache]: false,
+            [LoaderHeader.clientDetails]: {
+                capabilities: { interactive: false },
+                type: summarizerClientType,
+            },
+            [DriverHeader.summarizingClient]: true,
+            [LoaderHeader.reconnect]: false,
+            [LoaderHeader.version]: summaryVersion,
+        },
+        url: absoluteUrl,
+    };
+    const summarizerContainer = await loader.resolve(request);
+    await waitForContainerConnection(summarizerContainer);
+
+    const fluidObject =
+        await requestFluidObject<FluidObject<ISummarizer>>(summarizerContainer, { url: "_summarizer" });
+    if (fluidObject.ISummarizer === undefined) {
+        throw new Error("Fluid object does not implement ISummarizer");
+    }
+    return fluidObject.ISummarizer;
+}
+
+export async function createSummarizerFromFactory(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    dataStoreFactory: IFluidDataStoreFactory,
+    summaryVersion?: string,
+    containerRuntimeFactoryType = ContainerRuntimeFactoryWithDefaultDataStore,
+): Promise<ISummarizer> {
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            summaryConfigOverrides: {
+                state: "disableHeuristics",
+                maxAckWaitTime: 10000,
+                maxOpsSinceLastSummary: 7000,
+                initialSummarizerDelayMs: 0,
+                summarizerClientElection: false,
+            },
+        },
+    };
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new containerRuntimeFactoryType(
+        dataStoreFactory,
+        [
+            [dataStoreFactory.type, Promise.resolve(dataStoreFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const loader = provider.createLoader(
+        [[provider.defaultCodeDetails, runtimeFactory]],
+        { configProvider: mockConfigProvider() },
+    );
+
+    return createSummarizerCore(container, loader, summaryVersion);
+}
+
+export async function createSummarizer(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    summaryVersion?: string,
+    gcOptions?: IGCRuntimeOptions,
+    disableIsolatedChannels?: true,
+): Promise<ISummarizer> {
+    const testContainerConfig: ITestContainerConfig = {
+        runtimeOptions: {
+            summaryOptions: {
+                initialSummarizerDelayMs: 0,
+                summaryConfigOverrides: {
+                    state: "disableHeuristics",
+                    maxAckWaitTime: 10000,
+                    maxOpsSinceLastSummary: 7000,
+                    initialSummarizerDelayMs: 0,
+                    summarizerClientElection: false,
+                 },
```

Comment:
should we share this block that is shared between  createSummarizerFromFactory and createSummarizer ?

Reply:
sure, that sounds good.